### PR TITLE
Send XCM to call the contract on Moonbase

### PR DESCRIPTION
**Pull request by imstar15**

### CODE REVIEW

File: src/moonbeam/test-moonbase-contract/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    // Check chain state
+    const data = await incrementer.methods.number().call();
+    console.log(`The current number stored is: ${data}`);
+};
+
+main().catch(console.error).finally(() => {
+    console.log('Reached the end of main() ...');
+    process.exit();
+});
```
Comment (chrisli30) [inline src/moonbeam/test-moonbase-contract/index.js:42]: @imstar15 Let’s add your instructions from Slack to the README.md at the root level.

File: src/common/utils.js [hunk 2]
```diff
@@ -131,9 +131,10 @@ export const getProxyAccount = (api, sourceParaId, address, { addressType } = {
         ...multilocation.toU8a(),
     ]);
 
-    const DescendOriginAddress32 = u8aToHex(api.registry.hash(toHash).slice(0, 32));
-
-    return DescendOriginAddress32;
+    return {
+        accountId32: u8aToHex(api.registry.hash(toHash).slice(0, 32)),
+        accountKey20: u8aToHex(api.registry.hash(toHash).slice(0, 20)),
+    };
```
Comment (chrisli30) [inline src/common/utils.js:137]: 👍

**Review by chrisli30 — APPROVED** [review]
Local testing passed with Moonbase runtime 2201. Good to merge.

**Review by chrisli30 — CHANGES_REQUESTED** [review]
这代码质量标准太低了，问题包括但不限于，
各种hard-code
代码复用严重
代码顺序混乱，上下文没联系的代码堆在一起
变量命名歧义导致逻辑混乱，为什么图灵上的wallet叫moonbaseKeyPair？

**Comment by chrisli30** [comment]
Rebased master with a force push.

File: src/moonbeam/moonbase-alpha.js [hunk 3]
```diff
@@ -22,26 +22,112 @@ const TURING_INSTRUCTION_WEIGHT = 1000000000;
 
 const WEIGHT_PER_SECOND = 1000000000000;
 
+const CONTRACT_ADDRESS = '0xa72f549a1a12b9b49f30a7f3aeb1f4e96389c5d8';
```
[the remark this answers is not in the document]
Comment (imstar15) [reply src/moonbeam/moonbase-alpha.js:25]: Yes!
Please use the Incrementer contract address.

File: src/moonbeam/test-moonbase-contract/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (imstar15) [reply src/moonbeam/test-moonbase-contract/index.js:42]: It used to test the Increment contract.

I'll add some text to the README to explain it.

**Usage:**

```
CONTRACT_ADDRESS=<CONTRACT_ADDRESS> npm run test-moonbase-contract
```

File: src/moonbeam/moonbase-local.js [hunk 4]
```diff
@@ -165,7 +165,7 @@ const main = async () => {
     const turingAddressETH = aliceKeyPair.address;
     console.log('Turing wallet’s Ethereum address:', turingAddressETH);
 
-    const topUpAmount = (new BN(1000).mul(decimalBN)).toString();
+    const topUpAmount = (new BN(1000000).mul(decimalBN)).toString();
```
Comment (chrisli30) [inline src/moonbeam/moonbase-local.js:168]: What is the reason for the increment? We can’t create a proxy with 1000 DEV any more?

File: package.json [hunk 5]
```diff
@@ -8,16 +8,17 @@
...
     "lint": "eslint --fix src/*"
   },
   "dependencies": {
     "@inquirer/confirm": "^0.0.19-alpha.0",
     "@inquirer/select": "^0.0.18-alpha.0",
     "@mangata-finance/sdk": "^1.14.2",
-    "@oak-network/api-augment": "^1.8.0-rc.4",
-    "@oak-network/types": "^1.8.0-rc.4",
+    "@oak-network/api-augment": "^1.8.1-rc.1",
+    "@oak-network/types": "^1.8.1-rc.1",
```
Comment (chrisli30) [inline package.json:21]: @imstar15 We need to make these libraries official v1.8.1 and reference them.

**Review by chrisli30 — APPROVED** [review]

**Pull request merged**